example/udf/src/main/java/org/apache/iotdb/udf/table/MatchingKeyState.java

```diff
@@ -0,0 +1,623 @@
+  Set<String> negativePairs;
+  Map<Integer, Set<Integer>> distanceMap;
+  Map<String, int[]> distanceCache;
+  Set<String> CandidateCache;
+
+  public MatchingKeyState() {
+    reset();
+  }
+
+  public void init(int labelIndex, double min_support, double min_confidence) {
+    this.labelIndex = labelIndex;
+    this.min_support = min_support;
+    this.min_confidence = min_confidence;
+  }
+
+  public void add(int index, long time, String[] fullTuple) {
+    if (fullTuple.length <= labelIndex) {
+      throw new IllegalArgumentException("Tuple length is less than label index");
+    }
+    tupleList.add(new TupleEntry(index, time, fullTuple));
+  }
+
+  public Candidate computeAllPairs() {
+    for (int i = 0; i < tupleList.size(); i++) {
+      for (int j = i + 1; j < tupleList.size(); j++) {
+        totalPairs++;
+        TupleEntry t1 = tupleList.get(i);
+        TupleEntry t2 = tupleList.get(j);
+        int index = 0;
+        int[] distances = new int[columnLength];
+        int timeDiff = (int) Math.abs((t1.time - t2.time) / 1000);
+        distances[index] = timeDiff;
+        distanceMap.get(index).add(timeDiff);
+        for (int k = 0; k < columnLength; k++) {
+          if (k != labelIndex) {
+            index++;
+            int distance = editDistance(t1.tuple[k], t2.tuple[k]);
+            distances[index] = distance;
+            distanceMap.get(index).add(distance);
+          }
+        }
+        String label1 = t1.tuple[labelIndex];
+        String label2 = t2.tuple[labelIndex];
+        String key = i + "+" + j;
+        if (label1.equals(label2)) {
+          positivePairs.add(key);
+          distanceCache.put(key, distances);
+        } else {
+          negativePairs.add(key);
+          distanceCache.put(key, distances);
+        }
+      }
+    }
+    List<int[]> distanceRestrictions = new ArrayList<>();
+    for (int i = 0; i < distanceMap.size(); i++) {
+      Set<Integer> distances = distanceMap.get(i);
+      if (distances == null || distances.isEmpty()) {
+        distanceRestrictions.add(new int[] {0, 0});
+      } else {
+        int min = Collections.min(distances);
+        int max = Collections.max(distances);
+        distanceRestrictions.add(new int[] {min, max});
+        distanceMin[i] = min;
+        distanceMax[i] = max;
+      }
+    }
+    Candidate psi = new Candidate(distanceRestrictions, totalPairs);
+    for (String pair : positivePairs) {
+      psi.addpositive(pair);
+    }
+    for (String pair : negativePairs) {
+      psi.addNegative(pair);
+    }
+    return psi;
+  }
+
+  public void reset() {
+    tupleList = new ArrayList<>();
+    positivePairs = new HashSet<>();
+    negativePairs = new HashSet<>();
+    totalPairs = 0;
+    columnLength = 0;
+    distanceMap = new HashMap<>();
+    distanceCache = new HashMap<>();
+    CandidateCache = new HashSet<>();
+    labelIndex = 0;
+    min_support = 0.0;
+    min_confidence = 0.0;
+  }
+
+  public byte[] serialize() {
```

Comment:
add UT for serialize and deserialize method, make sure you can deserialize from what you serialize and the deserialized object is same as the previous one